File: generate-progress-report.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();
  
  // Set viewport for consistent rendering
  await page.setViewportSize({ width: 1400, height: 900 });
  
  // Load the HTML report
  await page.goto('http://localhost:3000/progress-report.html', { 
    waitUntil: 'networkidle',
    timeout: 30000 
  });
  
  // Wait for fonts and images to load
  await page.waitForTimeout(3000);
  
  // Generate PDF
  await page.pdf({
    path: '/app/public/Infinity_Legal_Progress_Report.pdf',
    format: 'A4',
    landscape: true,
    printBackground: true,
    margin: {
      top: '0',
      right: '0',
      bottom: '0',
      left: '0'
    }
  });
  
  console.log('PDF generated successfully: /app/public/Infinity_Legal_Progress_Report.pdf');
  
  await browser.close();
})();

File: generate-progress-report.py
import asyncio
from playwright.async_api import async_playwright

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        page = await browser.new_page()
        
        # Set viewport
        await page.set_viewport_size({"width": 1400, "height": 900})
        
        # Load the HTML report
        await page.goto('http://localhost:3000/progress-report.html', 
                        wait_until='networkidle',
                        timeout=30000)
        
        # Wait for fonts and images
        await page.wait_for_timeout(3000)
        
        # Generate PDF
        await page.pdf(
            path='/app/public/Infinity_Legal_Progress_Report.pdf',
            format='A4',
            landscape=True,
            print_background=True,
            margin={
                'top': '0',
                'right': '0',
                'bottom': '0',
                'left': '0'
            }
        )
        
        print('PDF generated: /app/public/Infinity_Legal_Progress_Report.pdf')
        await browser.close()

asyncio.run(main())
